test/MetaStable.integration.spec.ts

```diff
@@ -0,0 +1,339 @@
+    const userDataEncoded = defaultAbiCoder.encode(abi, data);
+
+    const exitPoolRequest = {
+        assets,
+        minAmountsOut: assets.map(() => '0'),
+        userData: userDataEncoded,
+        toInternalBalance: false,
+    };
+    const query = await helpers.queryExit(
+        poolId,
+        AddressZero, // Not important for query
+        AddressZero,
+        exitPoolRequest
+    );
+    return query;
+}
+
+export async function querySingleTokenExit(
+    network: number,
+    poolId: string,
+    assets: string[],
+    bptAmountIn: string,
+    exitTokenIndex: number
+): Promise<
+    [BigNumber, BigNumber[]] & { bptIn: BigNumber; amountsOut: BigNumber[] }
+> {
+    const helpers = BalancerHelpers__factory.connect(
+        ADDRESSES[network].balancerHelpers,
+        provider
+    );
+    const EXACT_BPT_IN_FOR_ONE_TOKEN_OUT = 0; // Alternative is: BPT_IN_FOR_EXACT_TOKENS_OUT (No proportional)
+    const abi = ['uint256', 'uint256', 'uint256'];
+
+    const data = [EXACT_BPT_IN_FOR_ONE_TOKEN_OUT, bptAmountIn, exitTokenIndex];
+    const userDataEncoded = defaultAbiCoder.encode(abi, data);
+
+    const exitPoolRequest = {
+        assets,
+        minAmountsOut: assets.map(() => '0'),
+        userData: userDataEncoded,
+        toInternalBalance: false,
+    };
+    const query = await helpers.queryExit(
+        poolId,
+        AddressZero, // Not important for query
+        AddressZero,
+        exitPoolRequest
+    );
+    return query;
+}
+
+describe('MetaStable', () => {
+    context('test joins vs queryJoin', () => {
+        // Setup chain
+        before(async function () {
+            this.timeout(20000);
+
+            await provider.send('hardhat_reset', [
+                {
+                    forking: {
+                        jsonRpcUrl,
+                        blockNumber: 16447247,
+                    },
+                },
+            ]);
+
+            sor = setUp(networkId, provider);
+            await sor.fetchPools();
+        });
+        context('Joins', () => {
+            it('Join with many tokens', async () => {
+                const amountsIn = [
+                    parseFixed('0.0123', 6),
+                    parseFixed('0.0456', 6),
+                ];
+
+                const pools = await subgraphPoolDataService.getPools();
+                const pool = MetaStablePool.fromPool(pools[0]);
+                const bptOut = pool._calcBptOutGivenExactTokensIn(amountsIn);
+
+                const deltas = await queryJoin(
+                    networkId,
+                    testPool.id,
+                    testPool.tokensList,
+                    amountsIn.map((a) => a.toString())
+                );
+                console.log('bptOut       ', formatFixed(bptOut, 18));
+                console.log('delta bptOut ', formatFixed(deltas.bptOut, 18));
+                // TODO: check if it's ok to have a small difference - diff comes from divDown not matching between TS and SC math
```

Comment:
This very small difference is fine. Worth making an Asana ticket to investigate change in divDown.

Reply:
✅